pkg/operator/controller/crl/crl_configmap.go

```diff
@@ -0,0 +1,295 @@
+)
+
+// authorityKeyIdentifier is a certificate's authority key identifier.
+type authorityKeyIdentifier struct {
+	KeyIdentifier []byte `asn1:"optional,tag:0"`
+}
+
+// authorityKeyIdentifierOID is the ASN.1 object identifier for the authority
+// key identifier extension.
+var authorityKeyIdentifierOID = asn1.ObjectIdentifier{2, 5, 29, 35}
+
+// ensureCRLConfigmap ensures the client CA certificate revocation list
+// configmap exists for a given ingresscontroller if the ingresscontroller
+// specifies a client CA certificate bundle in which any certificates specify
+// any CRL distribution points.  Returns a Boolean indicating whether the
+// configmap exists, the configmap if it does exist, and an error value.
+func (r *reconciler) ensureCRLConfigmap(ctx context.Context, ic *operatorv1.IngressController, namespace string, ownerRef metav1.OwnerReference, haveClientCA bool, clientCAConfigmap *corev1.ConfigMap) (bool, *corev1.ConfigMap, error) {
+	haveCM, current, err := r.currentCRLConfigMap(ctx, ic)
+	if err != nil {
+		return false, nil, err
+	}
+
+	var oldCRLs map[string]*pkix.CertificateList
+	if haveCM {
+		if data, ok := current.Data["crl.pem"]; ok {
+			if crls, err := buildCRLMap([]byte(data)); err != nil {
+				log.Error(err, "failed to parse current client CA configmap", "namespace", current.Namespace, "name", current.Name)
+			} else {
+				oldCRLs = crls
+			}
+		}
+
+	}
+
+	var clientCAData []byte
+	if haveClientCA {
+		clientCABundleFilename := "ca-bundle.pem"
+		if data, ok := clientCAConfigmap.Data[clientCABundleFilename]; !ok {
+			return haveCM, current, fmt.Errorf("client CA configmap %s/%s is missing %q", clientCAConfigmap.Namespace, clientCAConfigmap.Name, clientCABundleFilename)
+		} else {
+			clientCAData = []byte(data)
+		}
+	}
+
+	wantCM, desired, err := desiredCRLConfigMap(ic, ownerRef, clientCAData, oldCRLs)
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to build configmap: %w", err)
+	}
+
+	switch {
+	case !wantCM && !haveCM:
+		return false, nil, nil
+	case !wantCM && haveCM:
+		if err := r.client.Delete(ctx, current); err != nil {
+			if !errors.IsNotFound(err) {
+				return true, current, fmt.Errorf("failed to delete configmap: %w", err)
+			}
+		} else {
+			log.Info("deleted configmap", "namespace", current.Namespace, "name", current.Name)
+		}
+		return false, nil, nil
+	case wantCM && !haveCM:
+		if err := r.client.Create(ctx, desired); err != nil {
+			return false, nil, fmt.Errorf("failed to create configmap: %w", err)
+		}
+		log.Info("created configmap", "namespace", desired.Namespace, "name", desired.Name)
+		return r.currentCRLConfigMap(ctx, ic)
+	case wantCM && haveCM:
+		if updated, err := r.updateCRLConfigMap(ctx, current, desired); err != nil {
+			return true, current, fmt.Errorf("failed to update configmap: %w", err)
+		} else if updated {
+			log.Info("updated configmap", "namespace", desired.Namespace, "name", desired.Name)
+			return r.currentCRLConfigMap(ctx, ic)
```

Comment:
Could `updateCRLConfigMap` also return the updated value which would prevent the additional get here?

Reply:
It could, but this is the pattern we follow in other `ensure` functions.  I'd rather be consistent here, and we can refactor this and other code in a follow-up.